backend/src/animal/animal.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, Inject } from '@nestjs/common';
+import {FindManyOptions, Repository} from 'typeorm';
+import {Animal} from './animal.entity';
+import {ANIMAL_REPOSITORY_TOKEN} from './constants';
+import {DeepPartial} from 'typeorm/common/DeepPartial';
+import {Log} from '../logger/logger';
+
+@Component()
+export class AnimalService {
+  constructor(
+    @Inject(ANIMAL_REPOSITORY_TOKEN) private readonly animalRepository: Repository<Animal>,
+    private log: Log
+  ) {}
+
+  // Create
+  async create(animalDto: Animal): Promise<Animal> {
+    const animal = new Animal();
+    Object.assign(animal, animalDto);
```

Comment:
Using `plainToClass` from `class-transformer` is a better idea - `routing-controllers` create instance of params type automatically, this is where `nest` sucks 😞